onnxruntime/contrib_ops/cuda/bert/attention_softmax.cu

```diff
@@ -0,0 +1,900 @@
+    output[index] = T(val);
+  }
+}
+
+template <typename T, unsigned TPB>
+__global__ void SoftmaxKernel(const int all_sequence_length,
+                              const int sequence_length,
+                              const T* add_before_softmax,
+                              const T* input,
+                              T* output) {
+  Softmax<T, TPB>(all_sequence_length, sequence_length, all_sequence_length, 0,
+                  add_before_softmax, input, output);
+}
+
+template <typename T>
+Status ComputeSoftmax(cudaStream_t stream,
+                      const int all_sequence_length,
+                      const int sequence_length,
+                      const int batch_size,
+                      const int num_heads,
+                      const T* add_before_softmax,
+                      const T* input,
+                      T* output,
+                      bool is_unidirectional) {
+  const dim3 grid(sequence_length * num_heads, batch_size, 1);
+
+  bool need_call_online_softmax = EnvVarSingleton::instance().is_online_soft_max_enabled();
+  if (!need_call_online_softmax) {
+    if (all_sequence_length <= 32) {
+      const int blockSize = 32;
+      SoftmaxKernelSmall<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output, is_unidirectional);
+    } else if (all_sequence_length <= 64) {
+      const int blockSize = 64;
+      SoftmaxKernelSmall<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output, is_unidirectional);
+    } else if (all_sequence_length <= 128) {
+      const int blockSize = 128;
+      SoftmaxKernelSmall<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output, is_unidirectional);
+    } else if (all_sequence_length <= 256) {
+      const int blockSize = 256;
+      SoftmaxKernelSmall<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output, is_unidirectional);
+    } else if (all_sequence_length <= 512) {
+      const int blockSize = 512;
+      SoftmaxKernelSmall<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output, is_unidirectional);
+    } else if (all_sequence_length <= 1024) {
+      const int blockSize = 1024;
+      SoftmaxKernelSmall<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output, is_unidirectional);
+    } else if (!is_unidirectional) {
+      const int blockSize = 1024;
+      SoftmaxKernel<T, blockSize><<<grid, blockSize, 0, stream>>>(
+          all_sequence_length, sequence_length, add_before_softmax, input, output);
+    } else {
+      need_call_online_softmax = true;
+    }
+  }
+
+  if (need_call_online_softmax) {
```

Comment:
nit: combine two if to 
```
if (need_call_online_softmax) { ...} else {...}
```

Reply:
Need keep original path first, and make it easy to compare with original logic.